### PR TITLE
Task 05 Redux

### LINKED ISSUE #6: Task 05 - Flux + Redux
- Use provided API endpoint
- Fetch data through actions
- Pass data to components with Redux
- Unit tests for actions and reducers
- Store application state offline. Use data from offline storage at startup. See [redux-persist](https://github.com/rt2zz/redux-persist)

### PR DESCRIPTION
**Pull request by nikolay-borzov**
resolves #6 

Used `reselect`, `redux-actions`, `redux-thunk`, `redux-devtools-extension`, `redux-logger`, `redux-devtools-extension`, `redux-mock-store` and `redux-persist`

### CODE REVIEW

File: src/pages/film/film-container.jsx [hunk 1]
```diff
@@ -19,103 +24,97 @@ import {
...
+const mapStateToProps = state => ({
+  film: selectors.film.film(state),
+  filmIsFetching: selectors.film.isFetching(state),
+  genre: selectors.relatedFilms.searchParams.search(state),
+  relatedFilms: selectors.relatedFilms.films.films(state),
+  relatedFilmsIsFetching: selectors.relatedFilms.films.isFetching(state),
+  relatedFilmsError: selectors.relatedFilms.films.error(state)
+})
+
+const mapDispatchToProps = dispatch =>
```
Comment (tsvetkovv) [inline src/pages/film/film-container.jsx]:
```javascript
const mapDispatchToProps = {
  fetchFilm,
  setRelatedFilmsSearchParams,
  fetchRelatedFilms
}
```

See: [Using mapDispatchToProps() Shorthand Notation](https://egghead.io/lessons/javascript-redux-using-mapdispatchtoprops-shorthand-notation)

**Review by tsvetkovv — APPROVED** [review]
Well done

File: src/pages/film/film-container.jsx (hunk 1 above)
Comment (nikolay-borzov) [reply src/pages/film/film-container.jsx]: I see, that's more convenient!

**Comment by tsvetkovv** [comment]
Looks like extra info sends on the server:

`GET http://react-cdp-api.herokuapp.com/movies?search=deadpool&searchBy=title&sortBy=release_date&sortOrder=desc&limit=15&_persist=%7B%22version%22:-1,%22rehydrated%22:true%7D`

**Comment by nikolay-borzov** [comment]
Great, looks like `redux-persist` add it's data to the store.

**Pull request merged**